src/util/types.ts

```diff
@@ -1110,6 +1110,13 @@ export interface BaseAxisProps {
   label?: string | number | ReactElement | object;
   /** The HTML element's class name */
   className?: string;
+  /**
+   * When the cursor selects a value on the axis which is not present on another dataset,
+   * allow the nearest value to be selected instead.
+   */
+  allowSelectNearestValue?: boolean;
+
+  categoricalDomain?: any;
```

Comment:
what is this and why is it needed? Can it be typed?

Reply:
categoricalDomain is an array of the min/max values of the axis. 

This looks like it is calculated if the axis is of type ```'number'```, or if the axis is a data.

(apparently this variable also gets set if for the axis the ```scale !== 'auto'```, but I am not sure I visualize what that would mean for non numerical/date data, which cannot be easily ordered.

Just thought about adding it in the types as it wasn't documented there yet. 

I am not sure if a refactor on axis is in the works, but this is definitely a variable which at first glance seems like it is not well defined.

Reply:
oh this was here before? sorry didn't see that 😅